lib/LoginStrategy.ts

```diff
@@ -239,4 +253,49 @@ export class LoginStrategy extends BaseStrategy {
       return acc
     }, [] as Claim[])
   }
-}
+
+  /**
+   * @description checks if the claimer is authenticated into the DID Document
+   *
+   * @param token
+   * @param payload
+   *
+   * @returns {string} issuer DID or null
+   */
+  async verifyDidClaim(token: string, payload: ITokenPayload): Promise<string> {
+
+    const didDocument = await this.didResolver.read(payload.iss)
+
+    if (this.isAuthorized(token, didDocument))
+      return payload.iss
+    return null 
+  }
+
+  private isAuthorized = (claimedToken: string,didDocument: IDIDDocument) : boolean => {
```

Comment:
I think there are some of good pieces here, but I would suggest the following algorithm for this function:

1. filter the DID document public keys for all of the keys that are also in the `authentication` section of the document. This provides the "authentication" publicKeys
2. For each authentication publicKey, run the `verifyEthersSignature` function from the ew-did-registry `jwt` package to see if it matches. If one of the publicKeys verifies, then return true.